FileShares/src/FileSharesWeb/FileShareHealthContributor.cs

```diff
@@ -0,0 +1,84 @@
+using System.Runtime.InteropServices;
+using Microsoft.Extensions.Options;
+using Steeltoe.Common.HealthChecks;
+using Steeltoe.Common.Net;
+using Steeltoe.Samples.FileSharesWeb.Models;
+
+namespace Steeltoe.Samples.FileSharesWeb;
+
+internal partial class FileShareHealthContributor(IOptionsMonitor<FileShareOptions> fileShareOptionsMonitor) : IHealthContributor
```

Comment:
It would be good practice to extract the native call into a separate static class named `NativeMethods`, so we don't need `partial` here. Extracting it from this file keeps the sample easy to read.